src/status_im/chat/events/input.cljs

```diff
@@ -20,474 +23,566 @@
+  (when-let [cmp-ref (get-in chat-ui-props [current-chat-id ref])]
+    {::focus-rn-component cmp-ref}))
+
+(defn update-text-selection
+  "Updates text selection in active chat input"
+  [{:keys [current-chat-id] :as db} selection]
+  (let [input-text            (get-in db [:chats current-chat-id :input-text])
+        command               (input-model/selected-chat-command db current-chat-id input-text)
+        new-db                (model/set-chat-ui-props db {:selection selection})
+        chat-parameter-box-fx (load-chat-parameter-box new-db (:command command))]
+    (cond-> {:db new-db}
+
+      chat-parameter-box-fx
+      (merge chat-parameter-box-fx)
+
+      (and (= selection (+ (count const/command-char)
+                           (count (get-in command [:command :name]))
+                           (count const/spacing-char)))
+           (get-in command [:command :sequential-params]))
+      (merge (chat-input-focus new-db :seq-input-ref)))))
+
+(defn select-chat-input-command
+  "Selects command + (optional) arguments as input for active chat"
+  [{:keys [current-chat-id chat-ui-props] :as db}
+   {:keys [prefill prefill-bot-db sequential-params name] :as command} metadata prevent-auto-focus?]
+  (let [fx (-> db
+               bots-events/clear-bot-db
+               (model/set-chat-ui-props {:show-suggestions?   false
+                                         :result-box          nil
+                                         :validation-messages nil
+                                         :prev-command        name})
+               (set-chat-input-metadata metadata)
+               (set-chat-input-text (str (chat-utils/command-name command)
+                                         const/spacing-char
+                                         (when-not sequential-params
+                                           (input-model/join-command-args prefill))))
+               update-suggestions
+               (as-> fx'
+                   (merge fx' (load-chat-parameter-box (:db fx') command))))]
+    (cond-> fx
+      prefill-bot-db (update :db bots-events/update-bot-db {:db prefill-bot-db})
+
+      (not (and sequential-params
+                prevent-auto-focus?))
+      (merge (chat-input-focus (:db fx) :input-ref))
+
+      sequential-params
+      (as-> fx'
+          (cond-> (update fx' :db
+                          set-chat-seq-arg-input-text
+                          (str/join const/spacing-char prefill))
+            (not prevent-auto-focus?)
+            (merge fx' (chat-input-focus (:db fx') :seq-input-ref)))))))
+
+(defn set-contact-as-command-argument
+  "Sets contact as command argument for active chat"
+  [db {:keys [bot-db-key contact arg-index]}]
+  (let [name    (str/replace (:name contact) (re-pattern const/arg-wrapping-char) "")
+        contact (select-keys contact [:address :whisper-identity :name :photo-path :dapp?])]
+    (-> (set-command-argument db arg-index name true)
+        (as-> fx
```

Comment:
What is the sense of using threading macro here?

Reply:
We could have some function here instead of `as->`, like 
```clojure
(defn magic-fn [fx]
  (merge fx (bots-events/set-in-bot-db
             (:db fx)
             {:path [:public (keyword bot-db-key)]
             :value contact}))
```
and then just
```clojure
(-> (set-command-argument db arg-index name true)
      magic-fn
...)
```
because now it looks a bit weird imo

Reply:
I find `as->` macro very readable, but of course, I can extract the logic into small helper fn, so we don't have to use it